src/soc/ibm/power9/tor.h

```diff
@@ -0,0 +1,446 @@
+	EX_L2_REPR = 222,
+	EX_L3_REFR_REPR = 223,
+
+	/* Core Chiplet Rings */
+	/* Common - apply to all Core instances */
+	EC_FUNC = 224,
+	EC_GPTR = 225,
+	EC_TIME = 226,
+	EC_MODE = 227,
+
+	/* Core Chiplet Rings */
+	/* EC0 - EC23 instance specific Ring */
+	EC_REPR = 228,
+
+	/* Values 229-230 unused */
+
+	/* Core Chiplet Rings */
+	/* ABIST engine mode */
+	EC_ABST = 231,
+
+	/* Additional rings for Nimbus DD2 */
+	EQ_ANA_BNDY_BUCKET_26 = 232,
+	EQ_ANA_BNDY_BUCKET_27 = 233,
+	EQ_ANA_BNDY_BUCKET_28 = 234,
+	EQ_ANA_BNDY_BUCKET_29 = 235,
+	EQ_ANA_BNDY_BUCKET_30 = 236,
+	EQ_ANA_BNDY_BUCKET_31 = 237,
+	EQ_ANA_BNDY_BUCKET_32 = 238,
+	EQ_ANA_BNDY_BUCKET_33 = 239,
+	EQ_ANA_BNDY_BUCKET_34 = 240,
+	EQ_ANA_BNDY_BUCKET_35 = 241,
+	EQ_ANA_BNDY_BUCKET_36 = 242,
+	EQ_ANA_BNDY_BUCKET_37 = 243,
+	EQ_ANA_BNDY_BUCKET_38 = 244,
+	EQ_ANA_BNDY_BUCKET_39 = 245,
+	EQ_ANA_BNDY_BUCKET_40 = 246,
+	EQ_ANA_BNDY_BUCKET_41 = 247,
+
+	/* EQ Inex ring bucket */
+	EQ_INEX_BUCKET_1      = 248,
+	EQ_INEX_BUCKET_2      = 249,
+	EQ_INEX_BUCKET_3      = 250,
+	EQ_INEX_BUCKET_4      = 251,
+
+	/* CMSK ring */
+	EC_CMSK = 252,
+
+	/* Perv PLL filter override rings */
+	PERV_PLL_BNDY_FLT_1   = 253,
+	PERV_PLL_BNDY_FLT_2   = 254,
+	PERV_PLL_BNDY_FLT_3   = 255,
+	PERV_PLL_BNDY_FLT_4   = 256,
+
+	/* MC OMI rings */
+	MC_OMI0_FURE          = 257,
+	MC_OMI0_GPTR          = 258,
+	MC_OMI1_FURE          = 259,
+	MC_OMI1_GPTR          = 260,
+	MC_OMI2_FURE          = 261,
+	MC_OMI2_GPTR          = 262,
+	MC_OMIPPE_FURE        = 263,
+	MC_OMIPPE_GPTR        = 264,
+	MC_OMIPPE_TIME        = 265,
+	/* Instance rings */
+	MC_OMIPPE_REPR        = 266,
+
+	NUM_RING_IDS = 267
+};
+
+/* List of groups of rings */
+enum ring_class {
+	RING_CLASS_NEST,       	// Common NEST rings except GPTR #G rings
+	RING_CLASS_GPTR_NEST,	// Common GPTR #G rings-NEST
+	RING_CLASS_GPTR_EQ,    	// Common GPTR #G rings-EQ
+	RING_CLASS_GPTR_EX,    	// Common GPTR #G rings-EX
+	RING_CLASS_GPTR_EC,    	// Common GPTR #G rings-EC
+	RING_CLASS_EQ,		// Common EQ rings
+	RING_CLASS_EX,		// Common EX rings
+	RING_CLASS_EC,		// Common EC rings
+	RING_CLASS_EQ_INS,     	// Instance EQ rings
+	RING_CLASS_EX_INS,     	// Instance EX rings
+	RING_CLASS_EC_INS,     	// Instance EC rings
+};
+
+/* PPE types, enum values matches indecies inside rings section */
+enum ppe_type {
+	PT_SBE,
+	PT_CME,
+	PT_SGPE,
+};
+
+/* Available ring access operations */
+enum ring_operation {
+	GET_RING_DATA,
+	GET_RING_PUT_INFO,
+	GET_PPE_LEVEL_RINGS,
+};
+
+/* Result of calling tor_fetch_and_insert_vpd_rings() */
+enum ring_status {
+	RING_NOT_FOUND,
+	RING_FOUND,
+	RING_REDUNDANT,
+};
+
+/* Information necessary to put a ring into a ring section */
+struct ring_put_info {
+	uint32_t chiplet_offset;	// Relative to ring section
+	uint32_t ring_offset;           // Relative to ring section
+};
+
+/* Describes ring search characteristics for tor_fetch_and_insert_vpd_rings() */
+struct ring_query {
+	const char *kwd_name;		// Keyword name
```

Comment:
Is the length of name limited? If it is something sane (say, 8-16 characters or less), put an array here directly.
Also, as this structure is internal to code (not saved externally), please order the fields by size to keep padding low.